ledger/src/blockstore.rs

```diff
@@ -2915,46 +2918,70 @@ impl Blockstore {
                                 lookups.iter().for_each(|lookup| {
-                                    result.insert(lookup.account_key);
+                                    lookup_tables.insert(lookup.account_key);
                                 });
                             }
-                            // howdy, anybody who reached here from the panic messsage!
-                            // the .unwrap() below could indicate there was an odd error or there
-                            // could simply be a tx with a new ALT, which is just created/updated
-                            // in this range. too bad... this edge case isn't currently supported.
-                            // see: https://github.com/solana-labs/solana/issues/30165
-                            // for casual use, please choose different slot range.
-                            let sanitized_tx = bank.fully_verify_transaction(tx).unwrap();
-                            sanitized_tx
-                                .message()
-                                .account_keys()
-                                .iter()
-                                .for_each(|&pubkey| {
-                                    result.insert(pubkey);
-                                });
+
+                            let tx = SanitizedVersionedTransaction::try_from(tx)
+                                .expect("transaction failed to sanitize");
+                            let alt_scan_result =
+                                transaction_address_lookup_table_scanner::scan_transaction(&tx);
+                                match alt_scan_result {
+                                    transaction_address_lookup_table_scanner::ScanResult::NotFound => {},
+                                    transaction_address_lookup_table_scanner::ScanResult::NativeUsed(keys) => {
+                                        keys.into_iter().for_each(|key| { result.insert(key); });
+                                    }
+                                    transaction_address_lookup_table_scanner::ScanResult::NonNativeUsed(keys) => {
+                                        keys.into_iter().for_each(|key| { result.insert(key); });
+                                        possible_cpi_alt_extend.store(true, Ordering::Relaxed);
+                                    },
+                                }
                         });
                     });
                 }
             });
 
-        result
+        // For each unique lookup table add all accounts to the minimized set.
+        lookup_tables.into_par_iter().for_each(|lookup_table_key| {
+            let Some(lookup_table_account) = bank.get_account(&lookup_table_key) else {
+                return;
+            };
+
+            let Ok(lookup_table) = AddressLookupTable::deserialize(lookup_table_account.data())
+            else {
+                return;
+            };
+
+            for address in lookup_table.addresses.iter() {
```

Comment:
nit: mixed use of `for` loops and `.for_each()` in this pr to do `result.insert(...)`. maybe stick to one of the two?